### PR TITLE
zsh: portability improvements

### PR DESCRIPTION
**Pull request by r-darwish**

### CODE REVIEW

File: zsh/zshrc [hunk 1]
```diff
@@ -193,15 +195,18 @@ export EDITOR="/usr/bin/emacsclient -c"
 export ALTERNATE_EDITOR="/usr/bin/vim"
 export LESS="-R -n -X -m -i -S"
 
+if [[ ! -v ZSH_NO_COREUTILS ]]; then
```
Comment (drrlvn) [inline zsh/zshrc]: Use `-z`, it would be simpler. Also, you're missing a `$`.

File: zsh/zshrc [hunk 2]
```diff
@@ -268,20 +273,27 @@ zplug zsh-users/zsh-history-substring-search
...
+    source $ZPLUG_HOME/repos/junegunn/fzf/shell/key-bindings.zsh
+    export FZF_TMUX=1
+    export FZF_COMPLETION_TRIGGER="~~"
+    export FZF_DEFAULT_COMMAND='fd -t f'
+    export FZF_DEFAULT_OPTS="-e"
+    export FZF_CTRL_T_COMMAND="$FZF_DEFAULT_COMMAND"
+    export FZF_ALT_C_COMMAND="fasd -Rdl"
+    export FZF_ALT_C_OPTS="--preview 'tree --dirsfirst -CFh -L 2 {} | head -n 100'"
+    _fzf_compgen_path() {
+        if (( $+commands[fd] )); then
```
Comment (drrlvn) [inline zsh/zshrc]: The `if` should be around the function definition.

File: zsh/zshrc [hunk 3]
```diff
@@ -268,20 +273,27 @@ zplug zsh-users/zsh-history-substring-search
...
-export FZF_COMPLETION_TRIGGER="~~"
-export FZF_DEFAULT_COMMAND='fd -t f'
-export FZF_DEFAULT_OPTS="-e"
-export FZF_CTRL_T_COMMAND="$FZF_DEFAULT_COMMAND"
-export FZF_ALT_C_COMMAND="fasd -Rdl"
-export FZF_ALT_C_OPTS="--preview 'tree --dirsfirst -CFh -L 2 {} | head -n 100'"
-_fzf_compgen_path() {
-    fd -t f "$1"
-}
+if [[ -f $ZPLUG_HOME/repos/junegunn/fzf/shell/completion.zsh ]]; then
```
Comment (drrlvn) [inline zsh/zshrc]: Why is this needed? It should be installed with `zplug`.

File: zsh/zshrc [hunk 4]
```diff
@@ -268,20 +273,27 @@ zplug zsh-users/zsh-history-substring-search
...
+    source $ZPLUG_HOME/repos/junegunn/fzf/shell/completion.zsh
+    source $ZPLUG_HOME/repos/junegunn/fzf/shell/key-bindings.zsh
+    export FZF_TMUX=1
+    export FZF_COMPLETION_TRIGGER="~~"
+    export FZF_DEFAULT_COMMAND='fd -t f'
```
Comment (drrlvn) [inline zsh/zshrc]: This should probably also be inside the `if` for `fd`.

File: zsh/zshrc [hunk 5]
```diff
@@ -268,20 +273,27 @@ zplug zsh-users/zsh-history-substring-search
...
+    export FZF_DEFAULT_OPTS="-e"
+    export FZF_CTRL_T_COMMAND="$FZF_DEFAULT_COMMAND"
+    export FZF_ALT_C_COMMAND="fasd -Rdl"
```
Comment (drrlvn) [inline zsh/zshrc]: Should be inside the `if` for `fasd`.

File: zsh/zshrc (hunk 3 above)
Comment (r-darwish) [reply zsh/zshrc]: It refuses to be installed on platforms in which there's no precompiled binary for FZF
